packages/next-swc/crates/next-build-test/src/lib.rs

```diff
@@ -0,0 +1,140 @@
+#![feature(future_join)]
+#![feature(min_specialization)]
+#![feature(arbitrary_self_types)]
+
+use std::str::FromStr;
+
+use anyhow::{Context, Result};
+use futures_util::{StreamExt, TryStreamExt};
+use next_api::{
+    project::{ProjectContainer, ProjectOptions},
+    route::{Endpoint, Route},
+};
+
+pub async fn main_inner(strat: Strategy, factor: usize, limit: usize) -> Result<()> {
+    register();
+
+    let mut file = std::fs::File::open("project_options.json").with_context(|| {
+        let path = std::env::current_dir()
+            .unwrap()
+            .join("project_options.json");
+        format!("loading file at {}", path.display())
+    })?;
+
+    let options: ProjectOptions = serde_json::from_reader(&mut file).unwrap();
+    let project = ProjectContainer::new(options);
+
+    tracing::info!("collecting endpoints");
+    let entrypoints = project.entrypoints().await?;
+
+    let routes = shuffle(entrypoints.routes.clone().into_iter());
+    let count = render_routes(routes, strat, factor, limit).await;
+    tracing::info!("rendered {} pages", count);
+
+    Ok(())
+}
+
+pub fn register() {
+    next_api::register();
+    include!(concat!(env!("OUT_DIR"), "/register.rs"));
+}
+
+#[derive(PartialEq, Copy, Clone)]
+pub enum Strategy {
+    Sequential,
+    Concurrent,
+    Parallel,
+}
+
+impl std::fmt::Display for Strategy {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Strategy::Sequential => write!(f, "sequential"),
+            Strategy::Concurrent => write!(f, "concurrent"),
+            Strategy::Parallel => write!(f, "parallel"),
+        }
+    }
+}
+
+impl FromStr for Strategy {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self> {
+        match s {
+            "sequential" => Ok(Strategy::Sequential),
+            "concurrent" => Ok(Strategy::Concurrent),
+            "parallel" => Ok(Strategy::Parallel),
+            _ => Err(anyhow::anyhow!("invalid strategy")),
+        }
+    }
+}
+
+pub fn shuffle<'a, T: 'a>(items: impl Iterator<Item = T>) -> impl Iterator<Item = T> {
+    use rand::{seq::SliceRandom, thread_rng};
+    let mut rng = thread_rng();
+    let mut input = items.collect::<Vec<_>>();
+    input.shuffle(&mut rng);
+    input.into_iter()
```

Comment:
I think it would be better to have a pseudo random with a fixed seed, so the pages are deterministic on repeated runs